**1) Из колоды в 52 карты извлекаются случайным образом 4 карты.**

**a) Найти вероятность того, что все карты – крести.**

In [1]:
from math import factorial
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k))) 

#### Вычислим m:

In [2]:
m=combinations(13, 4)
print(f'm = {m}')

m = 715


#### Вычислим n:

In [3]:
n=combinations(52, 4)
print(f'n = {n}')

n = 270725


**Найдем вероятность P (4 трефы)**

In [4]:
P=m/n
print(f'P(4 трефы) = {round(P,4)}')

P(4 трефы) = 0.0026


#### б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

In [5]:
n=combinations(52, 4)
print(f'n = {n}')

n = 270725


In [6]:
m=sum([combinations(4,1)*combinations(48,3),combinations(4,2)*combinations(48,2),combinations(4,3)*combinations(48,1),1])
print(f'm = {m}')

m = 76145


In [7]:
P=m/n
print(f'P(хотя бы 1 туз) = {round(P,4)}')

P(хотя бы 1 туз) = 0.2813


#### Так же можно использовать то, что сумма вероятностей противоположных событий P(A)+P(Ā)=1 → P(A)=1-P(Ā)

In [8]:
P2=1-combinations(48,4)/combinations(52,4)
print(f'P(хотя бы 1 туз) = {round(P2,4)}')

P(хотя бы 1 туз) = 0.2813


**Ответ:
    1 Задача**
    
а) P(4 трефы)≈0.0026≈0.26%;

б) P(хотя бы 1 туз)≈0.2813≈28.13%.

#### 2) На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

In [12]:
P=1/combinations(10,3)
print(f'P(открыть с первой попытки) = {round(P,4)}')

P(открыть с первой попытки) = 0.0083


**Ответ: 2 задача**
    
P(открыть с первой попытки)≈0.0083≈0.83%.

#### 3) В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

In [10]:
P=combinations(9,3)/combinations(15,3)
print(f'P(3 из 3-х окрашены) = {round(P,4)}')

P(3 из 3-х окрашены) = 0.1846


**Ответ: 3 задача**

P(3 из 3-х окрашены)≈0.1846≈18.46%.

#### 4) В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

In [11]:
P=1/combinations(100,2)
print(f'P(оба выигрышных = {round(P,4)})')

P(оба выигрышных = 0.0002)


**Ответ: 4 задача**

P(оба выигрышных)≈0.0002≈0.02%.